src/dirwatcher.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const fsPromises = require('fs/promises');
+const { EventEmitter } = require('events');
+
+class DirWatcher extends EventEmitter {
+  constructor() {
+    super();
+    this.directoryContents = {};
+    this.newDirectoryContents = {};
+    this.changedFiles = [];
+  }
+
+  async readDir(path) {
```

Comment:
if method doesn't use `this` you can safely make it static and call like `DirWatcher.readDir()`